libcudacxx/codegen/codegen.cpp

```diff
@@ -142,6 +144,7 @@ int main() {
             for(auto& cv: cv_qualifier) {
                 out << "template<class _Type, _CUDA_VSTD::__enable_if_t<sizeof(_Type)==" << sz/8 << ", int> = 0>\n";
                 out << "_LIBCUDACXX_DEVICE void __atomic_load_cuda(const " << cv << "_Type *__ptr, _Type *__ret, int __memorder, " << scopenametag(s.first) << ") {\n";
+                out << "    if (__cuda_load_weak_if_local(__ptr, __ret)) return;\n";
```

Comment:
This should be weak_if_local_or_const_or_grid_param, since: 

```
__constant__ cuda::atomic<int> x;
x.load(); // UB, should use weak load
```

and

```
__global__ void kernel(__grid_param__ const cuda::atomic<int> x) { 
   x.load();
}
```

have the same issue.